util.go

```diff
@@ -190,7 +191,9 @@ func makeCompoundMessage(msgs [][]byte) *bytes.Buffer {
 
 	// Add the message lengths
 	for _, m := range msgs {
-		binary.Write(buf, binary.BigEndian, uint16(len(m)))
+		if err := binary.Write(buf, binary.BigEndian, uint16(len(m))); err != nil {
+			log.Fatal(err)
+		}
```

Comment:
`log.Fatal` calls `os.Exit`, which is an extremely problematic side-effect.
Swallow the error instead.